xapian-applications/omega/docs/newformat.rst

```diff
@@ -0,0 +1,156 @@
+
+First of all, we need a mime type for the new file format. Omega uses mime types to identify the format of a file and handle it in a proper way. The official registry for is at http://www.iana.org/assignments/media-types/ but not all filetypes have a corresponding official mime-type. In that case, a de-facto standard "x-" prefixed mime-type often exists. A good way to look for one is to ask the ``file`` utility to identify a file (Omega uses the same library as file to identify files when it does not recognise the extension)::
+
+  file --mime-type example.fb2
+
+which responses::
+
+  example.fb2: text/xml
+
+Sometimes ``file`` just returns a generic answer (most commonly ``text/plain`` or ``application/octet-stream``) and occasionally it misidentifies a file. If that is the case, we can associate the file format extension with a particular mime type at 'mimemap.tokens'. If multiple extensions are used for a format (such as htm and html for HTML) then add an entry for each.
+
+When indexing a filename which has an extension in upper-case or mixed-case, omindex will check for an exact match for the extension, and if not found, it will force the extension to lower-case and try again, so just add the extension in lower-case unless different cases actually have different meanings.
+
+In this example, ``text/xml`` is too broadly so we can associate ``fb2`` to ``application/x-fictionbook+xml`` which is much more specific.
+
+Then, you have to decide if you will use a library or an external filter.
+
+Using a library:
+================
+
+For safety reasons, it is not allowed to directly add libraries to omega source code. It is possible that some libraries contain bugs that affect the correct operation of the program. Because of that, omega isolates external libraries in subprocess using modules ``worker`` and ``assistant``. These modules will take care of the safety measures and use handlers to get access to the libraries.
+
+To begin with, we need a library to extract information from the desired format. If many options are available, try to choose a library with an active community, that supports UTF-8 encoding and has a licence compatible with MIT/X license and GNU GPL version 2 and later. If UTF-8 output is not supported, you can convert the encoding using convert_to_utf8(replacing "ISO-8859-1" with the character set which is produced)::
+
+  convert_to_utf8(dump, "ISO-8859-1");
+
+Once you have the library and the mimetype, it is time to modify the code. We have to create a new handler, which is the process using by omindex to access to the library. In order to do it, we have to create a file ‘handler_yourlibrary.cc’ that includes ‘handler.h’ and gives a definition to function ``extract`` (there are some examples at xapian-applications/omega such as 'handler_poppler.cc'). For this, inside the function body you will use the library to get the necessary information and storage it in the corresponding arguments::
```

Comment:
Changes:
* `create a new handler, which is the process using by omnindex` => `create a new handler, which is the process used by omnindex`
* `storage it` => `store it`

Reply:
Sorry, I will fix it.